packages/core/src/Site/index.js

```diff
@@ -366,7 +366,7 @@ class Site {
     let footer;
     if (fs.existsSync(wikiFooterPath)) {
       logger.info(`Copied over the existing ${WIKI_FOOTER_PATH} file to the converted layout`);
-      footer = fs.readFileSync(wikiFooterPath, 'utf8');
+      footer = `\n${fs.readFileSync(wikiFooterPath, 'utf8')}`;
```

Comment:
What's this about? Possible to show screenshots of before/after for your changes?

Reply:
(Side note: This change is shown as outdated because `Site/index.js` has been converted to `Site/index.ts`.)

In order to remove unnecessary newlines before the footer in `default.md`, I included nunjucks whitespace removers into the `set` and `endset` tags, like this, in `Site/siteConvertLayout.njk`:

```html
{%- set defaultFooter -%}
...
{%- endset -%}
```

This will ensure proper behavior for the default footers:

```html
<footer>
  <!-- Support MarkBind by including a link to us on your landing page! -->
  <div class="text-center">
    <small>[Generated by {{MarkBind}}]</small>
  </div>
</footer>
```

But for custom footers, as we are stripping the newlines from before the `set` tags as well (there are two), it results in this behavior:

```html
<footer>Custom footer
</footer>
```

![image](https://user-images.githubusercontent.com/25716529/232264794-c69dc0fc-803b-4555-a003-0158e2731fde.png)

You can check this as well by pulling this branch, removing this line, and then updating the tests to see the results!

Adding the newline here will ensure that the current behavior is not modified, i.e. this will only affect the custom footer to have an extra newline, producing this:

```html
<footer>
  Custom footer
</footer>
```

Reply:
All in all, this change is quite minor, so if we are fine with leaving generated custom footers to be without the newline then I can revert this newline addition - the user can modify the footer later on anyways, since this is only on `init`. I initially added this line just to keep to the current behavior :+1:

Reply:
If there's a problem we should fix it, doesn't matter whether this is minor or not

Reply:
For the files/screenshots that you mentioned above, could you indicate the filename/where they are from? Are they from .md or .html?

Could you provide a summary of what was the original behavior/output, what's the behavior/output without this newline, what's the final output with all the changes that you have made? I'm confused reading the explanation above.

Reply:
Also, is this change fixing `Remove generation of blank lines after the navbar in default.md` ? So the blank lines after the navbar is due to the footer? That's seems confusing.

Reply:
I've listed out the changes and the reason for these changes in a separate comment, hopefully that should be more comprehensible.